File: CSharp/OpenTraceability.GDST/Events/GDSTILMD.cs
using OpenTraceability.GDST.Events.KDEs;
using OpenTraceability.Models.Events;
using OpenTraceability.Utility.Attributes;

namespace OpenTraceability.GDST.Events
{
    public class GDSTILMD : EventILMD
    {
        [OpenTraceabilityObject]
        [OpenTraceability(Constants.CBVMDA_NAMESPACE, "vesselCatchInformationList")]
        [OpenTraceabilityJson("cbvmda:vesselCatchInformationList")]
        public VesselCatchInformationList? VesselCatchInformationList { get; set; }
        
        [OpenTraceability(Constants.GDST_NAMESPACE, "broodstockSource")]
        [OpenTraceabilityJson("gdst:broodstockSource")]
        public string? BroodstockSource { get; set; }

        [OpenTraceability(Constants.GDST_NAMESPACE, "aquacultureMethod")]
        [OpenTraceabilityJson("gdst:aquacultureMethod")]
        public string? AquacultureMethod { get; set; }

        [OpenTraceability(Constants.GDST_NAMESPACE, "proteinSource")]
        [OpenTraceabilityJson("gdst:proteinSource")]
        public string? ProteinSource { get; set; }
    }
}

File: Java/OpenTraceability/src/main/java/opentraceability/gdst/events/GDSTILMD.java
package opentraceability.gdst.events;

import opentraceability.Constants;
import opentraceability.gdst.events.kdes.VesselCatchInformationList;
import opentraceability.models.events.EventILMD;
import opentraceability.utility.attributes.*;

public class GDSTILMD extends EventILMD {
    @OpenTraceabilityObjectAttribute
    @OpenTraceabilityAttribute(ns = Constants.CBVMDA_NAMESPACE, name = "vesselCatchInformationList")
    @OpenTraceabilityJsonAttribute(name = "cbvmda:vesselCatchInformationList")
    private VesselCatchInformationList vesselCatchInformationList;

    @OpenTraceabilityAttribute(ns = Constants.GDST_NAMESPACE, name = "broodstockSource")
    @OpenTraceabilityJsonAttribute(name = "gdst:broodstockSource")
    private String broodstockSource;

    @OpenTraceabilityAttribute(ns = Constants.GDST_NAMESPACE, name = "aquacultureMethod")
    @OpenTraceabilityJsonAttribute(name = "gdst:aquacultureMethod")
    private String aquacultureMethod;

    @OpenTraceabilityAttribute(ns = Constants.GDST_NAMESPACE, name = "proteinSource")
    @OpenTraceabilityJsonAttribute(name = "gdst:proteinSource")
    private String proteinSource;

    // Add constructor, getters, and setters here
}





